pkg/vectorize/acos/acos.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package acos
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"math"
+)
+
+var (
+	acosUint8   func([]uint8, []float64) LogResult
+	acosUint16  func([]uint16, []float64) LogResult
+	acosUint32  func([]uint32, []float64) LogResult
+	acosUint64  func([]uint64, []float64) LogResult
+	acosInt8    func([]int8, []float64) LogResult
+	acosInt16   func([]int16, []float64) LogResult
+	acosInt32   func([]int32, []float64) LogResult
+	acosInt64   func([]int64, []float64) LogResult
+	acosFloat32 func([]float32, []float64) LogResult
+	acosFloat64 func([]float64, []float64) LogResult
+)
+
+func init() {
+	acosUint8 = acosUint8Pure
+	acosUint16 = acosUint16Pure
+	acosUint32 = acosUint32Pure
+	acosUint64 = acosUint64Pure
+	acosInt8 = acosInt8Pure
+	acosInt16 = acosInt16Pure
+	acosInt32 = acosInt32Pure
+	acosInt64 = acosInt64Pure
+	acosFloat32 = acosFloat32Pure
+	acosFloat64 = acosFloat64Pure
+}
+
+type LogResult struct {
```

Comment:
This variable name doesn't seem appropriate in this context

Reply:
I'm sorry that I was so careless